File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/NetTest.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

public class NetTest extends SLMessage {
    public NetBlock NetBlock_Field = new NetBlock();

    public static class NetBlock {
        public int Port;
    }

    public NetTest() {
        this.zeroCoded = false;
    }

    public int CalcPayloadSize() {
        return 6;
    }

    public void Handle(SLMessageHandler sLMessageHandler) {
        sLMessageHandler.HandleNetTest(this);
    }

    public void PackPayload(ByteBuffer byteBuffer) {
        byteBuffer.putShort(-1);
        byteBuffer.put((byte) 1);
        byteBuffer.put((byte) 70);
        packShort(byteBuffer, (short) this.NetBlock_Field.Port);
    }

    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.NetBlock_Field.Port = unpackShort(byteBuffer) & 65535;
    }
}

File: Linkpoint/src/main/app_broken_backup/com/linkpoint/slproto/messages/NetTest.kt
package com.linkpoint.slproto.messages

import com.linkpoint.slproto.SLMessage
import java.nio.ByteBuffer

class NetTest : SLMessage {
    NetBlock NetBlock_Field = NetBlock()

    class NetBlock {
        Int Port
    }

    NetTest() {
        this.zeroCoded = false
    }

    fun CalcPayloadSize(): Int {
        return 6
    }

    fun Handle(SLMessageHandler sLMessageHandler)  {
        sLMessageHandler.HandleNetTest(this)
    }

    fun PackPayload(ByteBuffer byteBuffer)  {
        byteBuffer.putShort(-1)
        byteBuffer.put((Byte) 1)
        byteBuffer.put((Byte) 70)
        packShort(byteBuffer, (this as Short).NetBlock_Field.Port)
    }

    fun UnpackPayload(ByteBuffer byteBuffer)  {
        this.NetBlock_Field.Port = unpackShort(byteBuffer) & 65535
    }
}
